src/Microsoft.ML.Transforms/WhiteningTransform.cs

```diff
@@ -108,72 +115,95 @@ public bool TryUnparse(StringBuilder sb)
             }
         }
 
-        public sealed class ColInfoEx
+        public sealed class ColInfo
         {
+            public readonly string Input;
+            public readonly string Output;
             public readonly WhiteningKind Kind;
-            public readonly Float Epsilon;
+            public readonly float Epsilon;
             public readonly int MaxRow;
             public readonly bool SaveInv;
             public readonly int PcaNum;
-            public readonly VectorType Type;
 
-            public ColInfoEx(Column item, Arguments args, ColInfo info)
+            /// <summary>
+            /// Describes how the transformer handles one column pair.
+            /// </summary>
+            /// <param name="input">Name of the input column.</param>
+            /// <param name="output">Name of the column resulting from the transformation of <paramref name="input"/>. Null means <paramref name="input"/> is replaced. </param>
+            /// <param name="kind">Whitening kind (PCA/ZCA).</param>
+            /// <param name="eps">Scaling regularizer.</param>
+            /// <param name="maxRows">Max number of rows.</param>
+            /// <param name="saveInverse">Whether to save inverse (recovery) matrix.</param>
+            /// <param name="pcaNum">PCA components to retain.</param>
+            public ColInfo(string input, string output = null, WhiteningKind kind = Defaults.Kind, float eps = Defaults.Eps,
```

Comment:
I am not very sure what those parameters mean. Please add more details if you can find something in TLC's doc site or in the code. This comment is not blocking you from checking this PR in.

Reply:
In the next iteration I will add some more details about the parameters.

---
In reply to: [229524812](https://github.com/dotnet/machinelearning/pull/1452#discussion_r229524812) [](ancestors = 229524812)